includes/ziti/model_map.h

```diff
@@ -80,6 +78,35 @@ for (var(e,line) = model_map_iterator(map), k = model_map_it_key(var(e,line)), v
 
 #define MODEL_MAP_FOR(it, m) for(model_map_iter it = model_map_iterator(&(m)); (it) != NULL; (it) = model_map_it_next(it))
 
+typedef struct model_list_s {
+    struct model_list_impl_s *impl;
+} model_list;
+
+typedef void *model_list_iter;
+
+ZITI_FUNC size_t model_list_size(const model_list *l);
+
+ZITI_FUNC void model_list_append(model_list *l, void *el);
```

Comment:
should doc all this. will this realloc on every append? i could read the code but doc would be better